kubectl-cloudflow/cmd/deploy.go

```diff
@@ -81,32 +105,198 @@ the stored credentials.
 	deployOpts.cmd.Flags().StringVarP(&deployOpts.password, "password", "p", "", "docker registry password.")
 	deployOpts.cmd.Flags().BoolVarP(&deployOpts.passwordStdin, "password-stdin", "", false, "Take the password from stdin")
 	deployOpts.cmd.Flags().StringArrayVar(&deployOpts.volumeMounts, "volume-mount", []string{}, "Accepts a key/value pair separated by an equal sign. The key should be the name of the volume mount, specified as '[streamlet-name].[volume-mount-name]'. The value should be the name of an existing persistent volume claim.")
+	deployOpts.cmd.Flags().StringToIntVar(&deployOpts.replicas, "scale", map[string]int{}, "Accepts key/value pairs for replicas per streamlet")
 
 	rootCmd.AddCommand(deployOpts.cmd)
 }
 
 func (opts *deployOptions) deployImpl(cmd *cobra.Command, args []string) {
 	version.FailOnProtocolVersionMismatch()
 
-	imageRef := args[0]
+	if (cmd.Flag("image").Changed == true) && (cmd.Flag("blueprint").Changed == true) {
+		util.LogAndExit("%s", "Cannot specify both blueprint and image in command")
+	}
+	if (cmd.Flag("image").Changed == false) && (cmd.Flag("blueprint").Changed == false) {
+		util.LogAndExit("%s", "Need to specify either blueprint or image in command")
+	}
+	// if (cmd.Flag("image").Changed == true) && (cmd.Flag("scale").Changed == true) {
+	// util.LogAndExit("%s", "scale is supported only with blueprint based deployment")
+	// }
+	if cmd.Flag("image").Changed == true {
+		deployImage(cmd, opts, args)
+	} else {
+		deployBlueprint(cmd, opts, args)
+	}
+}
+
+func deployBlueprint(cmd *cobra.Command, opts *deployOptions, args []string) {
+	// read blueprint file and load its content
+	blueprintFile := cmd.Flag("blueprint").Value.String()
+	content, err := ioutil.ReadFile(blueprintFile)
+	if err != nil {
+		util.LogAndExit("%s", err.Error())
+	}
+	config, err := loadBlueprintConfigContent(string(content[:]))
+	if err != nil {
+		util.LogAndExit("%s", err.Error())
+	}
+	fmt.Printf("%s\n", config)
+
+	// get the images from the blueprint
+	images := config.GetNode("blueprint.images").GetObject().Items()
+
+	var imageRefs []deploy.ImageReference
+	var dockerRegistryURL string
+
+	for _, imageRef := range images {
+		ref, err := parseImageReference(imageRef.GetString())
+		if err != nil {
+			util.LogAndExit("%s", err.Error())
+		}
+		imageRefs = append(imageRefs, *ref)
+		dockerRegistryURL = ref.Registry
+	}
+
+	applicationSpec, pulledImages := deploy.GetCloudflowApplicationSpecFromDockerImages(imageRefs, config, opts.replicas)
+	jsonb, err := json.Marshal(applicationSpec)
+	fmt.Println(string(jsonb))
+
+	namespace := applicationSpec.AppID
+	k8sClient, k8sErr := k8s.GetClient()
+	if k8sErr != nil {
+		util.LogAndExit("Failed to create new kubernetes client for Cloudflow application `%s`, %s", namespace, k8sErr.Error())
+	}
+
+	cloudflowApplicationClient, err := k8s.GetCloudflowApplicationClient(namespace)
```

Comment:
Thinking about it the idea of a CloudflowApplicationClient is a bit weird, it is just a k8s client right?

Reply:
yeah, k8s client for the cloudflow application - I kept the name that was used before.

Reply:
ok we can fix this in cleanup of code in separate PR.

Reply:
It's also weird that we create a k8sclient and then we need another client, which talks to k8s, but this is somehow a cloudflowApplicationClient. The code for creating those two is weird. We can delay to pick this up on cleanup. But have a look at what happens there.